### PR TITLE
Shamadi node draning fix

### PR DESCRIPTION
**Pull request by HamadiSE**

### CODE REVIEW

**Review by wying3 — APPROVED** [review]
tested in both clusters, e.g., cnfdt12 and cnfcert 3nodes cluster , but pass the test.

File: test-network-function/lifecycle/suite.go [hunk 1]
```diff
@@ -314,36 +314,36 @@ func testPodsRecreation(env *config.TestEnvironment) {
 			if len(deployments) == 0 {
 				return
 			}
+			if _, exists := namespaces[podNamespace]; !exists {
+				namespaces[podNamespace] = true
+			}
```
Comment (greyerof) [inline test-network-function/lifecycle/suite.go:319]: You'll repeat the getDeploymentsNode(podNamespace) on line 326 for each pod in the same namespace. Is it necessary? I suggest adding here this:
```
else {
    continue
}
```

Comment (HamadiSE) [reply test-network-function/lifecycle/suite.go:319]: right. will fix it

**Review by wying3 — APPROVED** [review]
reapprove

**Review by greyerof — APPROVED** [review]

**Pull request merged**